### DISCUSSION TITLE
Extend shared links with filters

### DISCUSSION THREAD

**Discussion by Cherry** [plausible/analytics#1495] [Anything Else]
[Shared links](https://plausible.io/docs/shared-links) are great, but sometimes you need to share a subset of stats. This could include:

- Stats for a particular utm campaign/partner 
- Stats for a particular country
- Stats for a particular directory/path

Being able to create "dashboards" or similar, or even just a shared link with a hard-set filter would be awesome to support this use-case.

**Comment by ukutaht** [comment]
Agreed.

I think first we should introduce some sort of 'saved filters' feature where you can have named, pre-made combinations of filters. Then one could attach a saved filter to a shared link to only share those stats

**Comment by jrgong420** [comment]
This would be great feature to share specific filters directly from the dashboard. Useful to share for example only a few stats e.g. a client

**Comment by NinjutsuSunrise** [comment]
First... we love Plausible... awesome tool! Thank you!

To chime in on the "hard-set filter" idea here...

It would be awesome for each "Shared Link" to simply have a "static" option. If on... then the ability to 1) X out applied filters 2) change filters or 3) download exports would be off. (simple UI removals tied to the shared link token)

We're testing implementing https://plausible.io/docs/embed-dashboard currently for folks that should only see a report for targeted custom property name=value pairs, but the full access to the rest of our data at just a couple clicks is kind of a blocker.

We understand the ability to change the filters via params would still be there... but for us this option would likely be "good enough" as it would limit the real problem features. In fact, being able to continue to change the filters via params eases our implementation.

**Reply by NinjutsuSunrise** [reply]
We ended up putting _superficial_ UI prevention in place. Obviously, anyone with intent can get around this, but for us this was "good enough". It would definitely be better to have something that removed the UI elements based on a server side setting tied to the Shared Link token.

CSS

```
iframe[src^="https://plausible.io/"] {
	pointer-events: none;
}
```

JS

```
setInterval(() => {

	const ele = document.querySelector('iframe[src^="https://plausible.io/"]')

	if (ele) {
		const com = window.getComputedStyle(ele)
		const val = com.getPropertyValue('pointer-events')
		if (val !== 'none') ele.style.pointerEvents = 'none'
	}

}, 500)
```
